### PR TITLE
Wait for dapr api to be ready to establish connection from dapr runtime.

### PR DESCRIPTION
**Pull request by kumaya**
# Description
- add blocking connection call with timeout.
- dapr runtime will crash after defined timeout. In case of k8s, the pod will be restarted post crash by default.

## Issue reference
Please reference the issue this PR will close: #5142 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/operator/client/client.go [hunk 1]
```diff
@@ -42,9 +44,12 @@ func GetOperatorClient(address, serverName string, certChain *daprCredentials.Ce
 	if err != nil {
 		return nil, nil, errors.Wrap(err, "failed to create tls config from cert and key")
 	}
-	opts = append(opts, grpc.WithTransportCredentials(credentials.NewTLS(config)))
+	// block for connection
+	opts = append(opts, grpc.WithTransportCredentials(credentials.NewTLS(config)), grpc.WithBlock())
 
-	conn, err := grpc.Dial(address, opts...)
+	ctx, cancelFunc := context.WithTimeout(context.Background(), 30*time.Second)
```
Comment (ItalyPaleAle) [inline pkg/operator/client/client.go:50]: Nit: `30 * time.Second` should be a constant (I don't think we need to make this configurable, yet)

**Review by ItalyPaleAle — DISMISSED** [review]

File: pkg/operator/client/client.go (hunk 1 above)
Comment (artursouza) [reply pkg/operator/client/client.go:50]: Any updates here?

Comment (kumaya) [reply pkg/operator/client/client.go]: Addressed the changes.

**Comment by kumaya** [comment]
any updates on this ?

**Review by artursouza — APPROVED** [review]

**Pull request merged**